Homework #1 Group Members: Soumya Agrawal, Parthiv Borgohain, Kshitij Mahajan

Question #1

In [2]:
"""calling the random module"""
import random
from sys import exit

"""Define the Global Variables"""
games_played = 0
user_wins = 0
count_of_rocks = 0
count_of_papers = 0
count_of_scissors = 0


"""If the computer chooses rock"""
def computer_action_rock():
    computer_action = "rock"
    user_action = input("Please Choose: rock, paper, scissors, or exit: ")
    global user_wins
    global games_played
    global count_of_rocks
    global count_of_papers
    global count_of_scissors
    
    if user_action == "rock":
        games_played += 1
        count_of_rocks += 1
        print ("computer chose", computer_action)
        print ("It's a tie")
        game()
    elif user_action == "paper":
        user_wins += 1
        games_played += 1
        count_of_papers += 1
        print ("computer chose", computer_action)
        print ("paper Covers rock, you WIN")
        game()
    elif user_action == "scissors":
        games_played += 1
        count_of_scissors += 1
        print ("computer chose", computer_action)
        print ("rock smashes scissors, you LOSE")
        game()
    elif user_action == "exit":
        print ("Thank you for playing! We played" ,games_played, "games_played, and won", user_wins, "of them")
        exit(0)
    else:
        print (user_action + " invalid selection")
        game()
        
"""If the computer chooses paper"""        
def computer_action_paper():
    computer_action = "paper"
    user_action = input("Please Choose: rock, paper, scissors, exit: ")
    global user_wins
    global games_played
    global count_of_rocks
    global count_of_papers
    global count_of_scissors
    
    if user_action == "rock":
        games_played += 1
        count_of_rocks += 1
        print ("computer chose", computer_action)
        print ("paper Covers rock, you LOSE")
        game()
    elif user_action == "paper":
        games_played += 1
        count_of_papers += 1
        print ("computer chose", computer_action)
        print ("It's a tie")
        game()
    elif user_action == "scissors":
        user_wins += 1
        games_played += 1
        count_of_scissors += 1
        print ("computer chose", computer_action)
        print ("scissors cut paper, you WIN")
        game()
    elif user_action == "exit":
        print ("Thank you for playing! We played" ,games_played, "games_played, and won", user_wins, "of them")
        exit(0)
    else:
        print (user_action, " invalid selection")
        game()
        
"""If the computer chooses scissors"""
def computer_action_scissors():
    computer_action = "scissors"
    user_action = input("Please Choose: rock, paper, scissors, exit: ")
    global user_wins
    global games_played
    global count_of_rocks
    global count_of_papers
    global count_of_scissors
    
    if user_action == "rock":
        user_wins += 1
        games_played += 1
        count_of_rocks += 1
        print ("computer chose", computer_action)
        print ("rock smashes scissors, you WIN")
        game()
    elif user_action == "paper":
        games_played += 1
        count_of_papers += 1
        print ("computer chose", computer_action)
        print ("scissors cut paper, you LOSE")
        game()
    elif user_action == "scissors":
        games_played += 1
        count_of_scissors += 1
        print ("computer chose", computer_action)
        print ("It's a tie")
        game()
    elif user_action == "exit":
        print ("Thank you for playing! We played" ,games_played, "games_played, and won", user_wins, "of them")
        exit(0)
    else:
        print (user_action + " invalid selection")
        game()

"""randomizing first game"""
def first_game():
        global user_wins
        global games_played
        global count_of_rocks
        global count_of_papers
        global count_of_scissors
        
        possible_actions = ["rock", "paper", "scissors"]
        computer_action = random.choice(possible_actions)
        user_action = input("Please Choose: rock, paper, scissors, exit: ")
        
        if user_action == computer_action:
            print(f"Both players selected {user_action}. It's a tie!")
            
            if user_action == "rock":
                count_of_rocks += 1
            elif user_action == "paper":
                count_of_papers += 1
            elif user_action == "scissors":
                count_of_scissors += 1
                
            games_played += 1
                
        elif user_action == "rock":
            count_of_rocks += 1
            if computer_action == "scissors":
                user_wins += 1
                print ("computer chose", computer_action)
                print("rock smashes scissors! You win!")
            elif computer_action == "paper":
                print ("computer chose", computer_action)
                print("paper covers rock! You lose.")
            games_played += 1
                
        elif user_action == "paper":
            games_played += 1
            count_of_papers += 1
            
            if computer_action == "rock":
                user_wins += 1
                print ("computer chose", computer_action)
                print("paper covers rock! You win!")
            elif computer_action == "scissors":
                print ("computer chose", computer_action)
                print("scissors cuts paper! You lose.")
            games_played += 1
                
        elif user_action == "scissors":
            games_played +=1
            count_of_scissors += 1
            if computer_action == "paper":
                user_wins += 1
                print ("computer chose", computer_action)
                print("scissors cuts paper! You win!")
            elif computer_action == "rock":
                print ("computer chose", computer_action)
                print("rock smashes scissors! You lose.")
            games_played += 1
        elif user_action == "exit":
            print ("Thank you for playing! We played" ,games_played, "games_played, and won", user_wins, "of them")
            exit(0)
        else:
            print (user_action + " invalid selection")
        
        

"""the game"""
def game():
    global user_wins
    global games_played
    global count_of_rocks
    global count_of_papers
    global count_of_scissors
    
    while games_played < 5:
        first_game()
                
    while games_played >= 5:        
        if count_of_scissors > count_of_rocks and  count_of_scissors > count_of_papers:
            computer_action_rock()
        elif count_of_papers > count_of_rocks and count_of_papers > count_of_scissors:
            computer_action_scissors()
        elif count_of_rocks > count_of_papers and count_of_rocks > count_of_scissors:
            computer_action_paper()
        elif count_of_papers > count_of_rocks and count_of_rocks == count_of_scissors:
            computer_action_scissors()
        elif count_of_rocks > count_of_papers and count_of_papers == count_of_scissors:
            computer_action_paper()
        elif count_of_scissors > count_of_papers and count_of_papers == count_of_rocks:
            computer_action_rock()
        elif count_of_papers==count_of_rocks and count_of_rocks==count_of_scissors:
            computer_action = random.choice(possible_actions)
        elif count_of_papers==count_of_rocks and count_of_rocks > count_of_scissors:
            computer_action_paper()
        elif count_of_papers==count_of_scissors and count_of_scissors > count_of_rocks:
            computer_action_scissors()
        elif count_of_rocks==count_of_scissors and count_of_scissors > count_of_papers:
            computer_action_rock()
            
        
"""This is calling the original function to play the game"""        
game()


Please Choose: rock, paper, scissors, exit: paper
computer chose rock
paper covers rock! You win!
Please Choose: rock, paper, scissors, exit: rock
computer chose paper
paper covers rock! You lose.
Please Choose: rock, paper, scissors, exit: scissors
Both players selected scissors. It's a tie!
Please Choose: rock, paper, scissors, exit: paper
computer chose rock
paper covers rock! You win!
Please Choose: rock, paper, scissors, exit: paper
computer chose scissors
scissors cut paper, you LOSE
Please Choose: rock, paper, scissors, exit: paper
computer chose scissors
scissors cut paper, you LOSE
Please Choose: rock, paper, scissors, exit: rock
computer chose scissors
rock smashes scissors, you WIN
Please Choose: rock, paper, scissors, exit: scissors
computer chose scissors
It's a tie
Please Choose: rock, paper, scissors, exit: paper
computer chose scissors
scissors cut paper, you LOSE
Please Choose: rock, paper, scissors, exit: paper
computer chose scissors
scissors cut paper, you LOSE
Plea

SystemExit: 0

Question #2

In [4]:
import re
count=0
DataList=[]
# Using readlines() to read th file line by line and storing lines in a list
file1 = open('Datasets/FloridaVoters.html', 'r')
Lines = file1.readlines()
#Used exception handler like try-except block to pass an attribute error. Reference:- pythontool website
# The required text blocks are present between <td> elements of a HTML file. Hence, used search operation of regular 
#expression to find the exact text block from a line in List.
for line in Lines:
        try:
            Data= re.search('<td>([\w,-.\s]+)</td>', line).group(1)
            Data = Data.replace(',', '')
            DataList.append(Data)
            count=count+1
        except AttributeError:
            pass
i=0

FinalList=[]
# We only need first three text elements of each text block containg six elements. 
#Hence, run the for loop with stepsize of 6
for k in range(0,int(count/6)):   
    try:
        if i==65:
            break
        newTuple=()
        #Store first three elements of each text block into a tuple.
        newTuple=(DataList[i],DataList[i+1],DataList[i+2])
        i=i+6
        #Make a list of tuples
        FinalList.append(newTuple)
    except IndexError:
        pass
#Sort the list on the basis of third element of each tuple in a list   
SortedList = sorted(FinalList, key = lambda x: int(x[2]))
    
for row in SortedList:
    print (row[0],row[1],row[2])

LAFAYETTE 1373 2672
GLADES 2190 3110
LIBERTY 720 3372
UNION 2752 3579
GILCHRIST 5789 3652
FRANKLIN 2234 4319
HOLMES 5282 4434
GULF 4234 4521
HARDEE 4859 4702
HAMILTON 2154 4796
DIXIE 3314 4839
CALHOUN 2201 5324
WASHINGTON 7101 5687
JEFFERSON 2636 5802
BAKER 6963 5813
BRADFORD 6878 6533
TAYLOR 3950 6915
MADISON 2992 7158
DESOTO 4870 7181
OKEECHOBEE 7755 7756
HENDRY 5862 7999
WAKULLA 7374 8889
LEVY 11665 9509
WALTON 25609 10013
SUWANNEE 10745 11126
NASSAU 32958 14013
COLUMBIA 15790 14797
JACKSON 9626 15706
MONROE 20602 17614
HIGHLANDS 27100 19997
PUTNAM 17067 20606
GADSDEN 4372 22161
SUMTER 47158 22977
FLAGLER 30047 24734
OKALOOSA 75154 25172
SANTA ROSA 73627 26114
MARTIN 53800 27358
INDIAN RIVER 47794 28204
CITRUS 46373 30440
BAY 57456 30733
CLAY 76584 31285
CHARLOTTE 54421 35602
ST. JOHNS 88385 39531
HERNANDO 51254 42499
COLLIER 100167 45511
LAKE 93604 67237
MANATEE 96063 67926
ESCAMBIA 90265 70180
OSCEOLA 44594 75657
ST. LUCIE 59626 76163
MARION 97306 76268
ALACHUA 47329 77996
SARASOT

Question #3

Part(a)

In [5]:
def moving_window(tweet):
    L1=[]                                       #initializing an empty list to store three-phrase words
    L2=[]                                       #initializing an empty list to store value 1
    for index in range(len(tweet)):             #creating a string of three words
        if index<=len(tweet)-3:
            string1=' '.join(tweet[index:index+3])
            L1.append(string1)                  #adding the three word strings to list L1
    for idx in range(len(L1)):
        L2.append(1)                            #adding 1 (value) to list L2
    tuple_phrases=tuple(zip(L1,L2))             #creating a tuple from Lists L1 and L2 (to convert to dictionary)
    phrases=dict((x,y) for x,y in tuple_phrases) #creating a dictionary 
    return phrases

Part (b)

In [6]:

def cosine(tweet1,tweet2):
    matches=0                                   #initializing matches to 0
    n1=len(moving_window(tweet1))               #initializing n1 to no of three word phrases in 1st tweet
    n2=len(moving_window(tweet2))               #initializing n2 to no of three word phrases in 1st tweet
    for key in moving_window(tweet1):
        if key in moving_window(tweet2):        
            matches=matches+1                   #incrementing matches if a same phrase is found in 2nd tweet
    x=matches/math.sqrt(n1*n2)                  #computing cosine function
    return x


Part (c)

In [8]:
import math
fp=open('Datasets/Santa.txt')                   #reading txt file
line = fp.readline()                            #reading 1st line of file
tweets_list=[]                                  #initializing an empty list to store tweets from file
while line != '':
    words=line.lstrip().rstrip().lstrip('""').rstrip('""').split(' ')
    line=fp.readline()
    tweets_list.append(words)                   #adding tweet to the the list tweets_list
fp.close()

#printing all tweets sequentially
print("List of Tweets:\n")
for index,tweet in enumerate(tweets_list):
    X="Tweet No." + str(index+1) + ": "
    Y=" ".join(tweet)
    print(X+Y)
    
    
#analyzing tweets one by one
for index1,tweet1 in enumerate(tweets_list[1:]):   #Loooping from 2nd element of tweets list
    for index2,tweet2 in enumerate(tweets_list):   #Looping from 1st element of tweets list so that we can compare a tweet to its previously seen tweets
        if index2>=index1+1:
            continue                             
        else:
            similarity=cosine(tweets_list[index2],tweets_list[index1+1])
            if similarity>=0.5:
                print("\nTweet No. ",index2+1,"is similar to Tweet No. ",index1+2,"!!")
                print(similarity)
                

List of Tweets:

Tweet No.1: SPECIAL SECRET HEARTS: A Child's Introduction to Dementia and Pink Curls - A Santa ... - http://t.co/UWCdc8FA9a http://t.co/meexKLGTKl
Tweet No.2: Santa Claus Is Coming To Town #MTVHottest Justin Bieber
Tweet No.3: RT @BuyBookstore: SPECIAL SECRET HEARTS: A Child's Introduction to Dementia and Pink Curls - A Santa ... - http://t.co/UWCdc8FA9a http://t.Â…
Tweet No.4: RT @DrewFtDevonne_: Rt si te gusta Santa Claus Is Coming To Town #MTVHottest Justin Bieber
Tweet No.5: Rt si te gusta Santa Claus Is Coming To Town #MTVHottest Justin Bieber

Tweet No.  1 is similar to Tweet No.  3 !!
0.8838834764831844

Tweet No.  2 is similar to Tweet No.  4 !!
0.7337993857053429

Tweet No.  2 is similar to Tweet No.  5 !!
0.7977240352174656

Tweet No.  4 is similar to Tweet No.  5 !!
0.9198662110077999


Question #4

Part(a)

In [9]:
import math
import re
newList2=[]
newList=[]
#Open a file in read mode)
file1 = open('Datasets/quotes.txt','r',encoding = "ISO-8859-1")
#Read lines in file and store them in a List
initialList = file1.readlines()
Lines=[]
for k in initialList:
    Lines.append(k.replace("\n",""))  #Remove newline tags from the list
#Run a for loop parsing through list with stepsize of 2 
#Used exception handling to avoid IndexError 
for i in range(0,len(Lines),2):
    try:
        string1=str('"'+Lines[i]+" "+'-')
        string2=str(Lines[i+1]+'"')
        string3=str(Lines[i]+" "+'-')
        string4=str(Lines[i+1])
        newList.append(string3+string4)  # Add quote and speaker
        newList2.append(string1+string2) # Add quote and speaker within inverted commas
    except IndexError:
        pass

for eachline in newList2:
    print(eachline)

"How we spend our days is, of course, how we spend our lives. -Annie Dillard"
"Two roads diverged in a wood, and I...I took the one less traveled by, and that has made all the difference. -Robert Frost"
"What is happiness? The feeling that power is growing, that resistance is overcome. -Friedrich Nietzsche"
"A great deal of intelligence can be invested in ignorance when the need for illusion is deep. -Saul Bellow"
"Those who are preoccupied with `making a statement` usually don`t have any statements worth making. -Thomas Sowell"
"Women need a reason to have sex -- men just need a place. -Billy Crystal"
"The heart has its reasons, of which the mind knows nothing. -Blaise Pascal"
"It is necessary for the welfare of society that genius should be privileged to utter sedition, to blaspheme, to outrage good taste, to corrupt the youthful mind, and generally to scandalize one`s uncles. -George Bernard Shaw"
"Freedom is not something that anybody can be given; freedom is something people take 

Part (b)

In [9]:
import re
quoteList=[]
for j in range(0,len(Lines),2):
    quoteList.append(Lines[j]) #Separated speaker from full-quotes

#print(quoteList)

def convertWords (m):
    words=[]
    words = re.split("\W",m) #split line into words such that words include characters,digits, and underscore only.
    words=(map(lambda x: x.lower(), words)) # Make sure that all elements of list are in lower cases only.
    words = list(words) 
    wordsnew=[]
    for j in words:
        wordsnew.append(j.replace("\n","")) #Remove newline tags from the list
    res = []
    for ele in wordsnew:
        if ele.strip(): #check if list contains any spaces. In split fucntion spaces are considered as words. To remove them used if block.
            res.append(ele) 
    return res
words=[] 

for h in range(0,len(newList),1):
    words=convertWords(str(newList[h]))
    print(words)

['how', 'we', 'spend', 'our', 'days', 'is', 'of', 'course', 'how', 'we', 'spend', 'our', 'lives', 'annie', 'dillard']
['two', 'roads', 'diverged', 'in', 'a', 'wood', 'and', 'i', 'i', 'took', 'the', 'one', 'less', 'traveled', 'by', 'and', 'that', 'has', 'made', 'all', 'the', 'difference', 'robert', 'frost']
['what', 'is', 'happiness', 'the', 'feeling', 'that', 'power', 'is', 'growing', 'that', 'resistance', 'is', 'overcome', 'friedrich', 'nietzsche']
['a', 'great', 'deal', 'of', 'intelligence', 'can', 'be', 'invested', 'in', 'ignorance', 'when', 'the', 'need', 'for', 'illusion', 'is', 'deep', 'saul', 'bellow']
['those', 'who', 'are', 'preoccupied', 'with', 'making', 'a', 'statement', 'usually', 'don', 't', 'have', 'any', 'statements', 'worth', 'making', 'thomas', 'sowell']
['women', 'need', 'a', 'reason', 'to', 'have', 'sex', 'men', 'just', 'need', 'a', 'place', 'billy', 'crystal']
['the', 'heart', 'has', 'its', 'reasons', 'of', 'which', 'the', 'mind', 'knows', 'nothing', 'blaise', 'pas

Part (c)

In [13]:
def word_count(n):  # Function to count the number times a word is used in a line.
    counts = dict()
    words=convertWords(n) # Convert input line into the list of words using previously written function.

    for word in words:
        if word in counts:
            counts[word] += 1
        else:
            counts[word] = 1

    return counts  #Return a dictionary having keys as words and values as the number of times specific word is used.

def postings_lists(b):
    newDic=dict()
    newDic2=word_count(b) #Store dictionary of word counts.
    newDic[str(b)]=newDic2 # Create a dictionary having key a quote and value a dictionary of word counts.
    
    return newDic
FinalDic=dict()
for b in range(0,len(newList),1):
    newDic=dict()
    newDic=postings_lists(newList[b])
    FinalDic=FinalDic | newDic      #Create a postings list dictionary for all quotes.

print(FinalDic)

def posting2(newList):  # Another posting list function with different input type. To be used in the coming sections.
    posting = {}
    for j in newList:
        words = convertWords(j)
        quote = {}
        for w in words:
            if w in quote:
                quote[w] += 1
            else: 
                quote[w] = 1
        posting[j] = quote
    return posting

{'How we spend our days is, of course, how we spend our lives. -Annie Dillard': {'how': 2, 'we': 2, 'spend': 2, 'our': 2, 'days': 1, 'is': 1, 'of': 1, 'course': 1, 'lives': 1, 'annie': 1, 'dillard': 1}, 'Two roads diverged in a wood, and I...I took the one less traveled by, and that has made all the difference. -Robert Frost': {'two': 1, 'roads': 1, 'diverged': 1, 'in': 1, 'a': 1, 'wood': 1, 'and': 2, 'i': 2, 'took': 1, 'the': 2, 'one': 1, 'less': 1, 'traveled': 1, 'by': 1, 'that': 1, 'has': 1, 'made': 1, 'all': 1, 'difference': 1, 'robert': 1, 'frost': 1}, 'What is happiness? The feeling that power is growing, that resistance is overcome. -Friedrich Nietzsche': {'what': 1, 'is': 3, 'happiness': 1, 'the': 1, 'feeling': 1, 'that': 2, 'power': 1, 'growing': 1, 'resistance': 1, 'overcome': 1, 'friedrich': 1, 'nietzsche': 1}, 'A great deal of intelligence can be invested in ignorance when the need for illusion is deep. -Saul Bellow': {'a': 1, 'great': 1, 'deal': 1, 'of': 1, 'intelligence':

Part (d)

In [11]:
FinalWordList=[]
for k in range(0,len(newList),1):
    words=convertWords(str(newList[k]))
    FinalWordList=FinalWordList+words    #Make a list of all words in the given file.

#print(FinalWordList)
#A reverse postings-list is a dictionary whose keys are the words, and the values are themselves
#dictionaries with the key being a full quote, and the value being the number of times the word 
#appeared in the full quote.
def reverse_postings_Lists(FinalWordList):
    
    reverseDic2=dict()
    for word in FinalWordList:
        reverseDic=dict()
        for line in newList:
            if word in convertWords(line):
                
                reverseDic[line]=convertWords(line).count(word)
            else:
                next
            reverseDic2[word]=reverseDic
    return reverseDic2

LastDic=dict()

LastDic=reverse_postings_Lists(FinalWordList)
print(LastDic)

{'how': {'How we spend our days is, of course, how we spend our lives. -Annie Dillard': 2, 'Christmas can be celebrated in the school room with pine trees, tinsel and reindeers, but there must be no mention of the man whose birthday is being celebrated. One wonders how a teacher would answer if a student asked why it was called Christmas. -Ronald Reagan': 1, 'I have often wondered how it is that every man loves himself more than all the rest of men, but yet sets less value on his own opinion of himself than on the opinion of others. -Marcus Aurelius': 1, 'It`s not how old you are, it`s how hard you work at it. -Jonah Barrington': 2, 'No matter how much cats fight, there always seem to be plenty of kittens. -Abraham Lincoln': 1, 'Success, the real success, does not depend upon the position you hold but upon how you carry yourself in that position. -Theodore Roosevelt': 1, 'Never discourage anyone who continually makes progress, no matter how slow. -Plato': 1, 'True wisdom comes to each 

part(e)

In [14]:
def tfidf(word,quote):
    if word not in posting2(newList)[quote]:
        tfdf = 0 
    
    else:
        count1=convertWords(quote).count(word) # number of times a word has appeared in the quote
        count2=max(posting2(newList)[quote].values()) #maximum number of times any word occurs in the quote.
        tf=float(count1/count2)
    
        count3=len(newList) #Number of quotes in the file.
        count4=len(reverse_postings_Lists(FinalWordList)[word].values())#number of full quotes that contain the word .
        idf=math.log(float(count3/count4))
        tfidf=tf*idf
    return(tfidf)
print(tfidf("entertainer", "An actor is at most a poet and at least an entertainer. -Marlon Brando"))

3.3933584753025405


part(f)

In [20]:
#A function that takes a word as argument, and returns a dictionary whose keys are 
#full quotes containing that word, and whose values are the TF-IDF score of that word 
#for that full quote.
def search_single(word):
    singleDic=dict()
    singleDic=LastDic[word]
    singleList=[]
    singleList=singleDic.keys()
    Dicforsingle=dict()
    for line in singleList:
        value=tfidf(word,line)
        Dicforsingle[line]=value
        
    return Dicforsingle
k=str("disease")
singleDic=dict()
singleDic=search_single(k)
print(singleDic)

{'Everything that used to be a sin is now a disease. -Bill Maher': 3.3933584753025405}


part(g)

In [ ]:
#a function that takes a list of words as argument, and returns a dictionary 
#whose keys are full quotes containing one or more of the words in that list, 
#and whose values are the sum of TF-IDF scores of the words in that list for that full quote.
def search_multiple(wordlist):
    DicwordList =dict() 
    for word in wordlist:
        DicFinal=dict()
        DicFinal=search_single(word)
        for key in DicFinal:
            if key in DicwordList:
                DicwordList[key] += DicFinal[key]
            else:
                DicwordList[key] = DicFinal[key]
    
    return DicwordList
outputDic=dict()
outputDic=search_multiple(['heart','mind','disease'])
print (outputDic.items())

Sample Output:
dict_items([('The heart has its reasons, of which the mind knows nothing. - Blaise Pascal', 0.5655597458837567), ('In every man`s heart there is a secret nerve that answers to the vibrations of beauty. - Christopher Morley', 0.6786716950605081), ('Everybody can be great... because anybody can serve. You don`t have to have a college degree to serve. You don`t have to make your subject and verb agree to serve. You only need a heart full of grace. A soul generated by love. - Martin Luther King Jr.', 0.16966792376512702), ('Love anything and your heart will be wrung and possibly broken. If you want to make sure of keeping it intact you must give it to no one, not even an animal. Wrap it carefully round with hobbies and little luxuries; avoid all entanglements. Lock it up safe in the casket or coffin of your selfishness. But in that casket, safe, dark, motionless, airless, it will change. It will not be broken; it will become unbreakable, impenetrable, irredeemable. To love is to be vulnerable. - C.S. Lewis', 0.09695309929435829), ('We all wish to be judged by our peers, by the men `after our own heart.` Only they really know our mind and only they judge it by standards we fully acknowledge. Theirs is the praise we really covet and the blame we really dread. The little pockets of early Chrstians survived because they cared exclusively for the love of `the bretheren` and stopped their ears to the opinion of the Pagan society around them. But a circle of criminals, cranks, or perverts survives in just the same way; by becoming deaf to the opinion of the outer world, by discounting it as the chatter of outsiders who `don`t understand,` of the `conventional,` the `bourgeois,` the `Establishment,` of prigs, prudes, and humbugs. - C.S. Lewis', 0.07540796611783424), ('I will honor Christmas in my heart, and try to keep it all the year. - Charles Dickens', 0.6786716950605081), ('Show me your hands. Do they have scars from giving? Show me your feet. Are they wounded in service? Show me your heart. Have you left a place for divine love? - Fulton J. Sheen', 0.22622389835350268), ('`Tis the business of little minds to shrink, but he whose heart is firm, and whose conscience approves his conduct, will pursue his principles unto death. - Thomas Paine', 0.33933584753025403), ('The best and most beautiful things in the world cannot be seen or even touched. They must be felt with the heart. - Helen Keller', 0.22622389835350268), ('The mystic chords of memory, stretching from every battlefield and patriot grave to every living heart and hearthstone all over this broad land, will yet swell the chorus of the Union, when again touched, as surely they will be, by the better angels of our nature. - Abraham Lincoln', 0.16966792376512702), ('It is necessary for the welfare of society that genius should be privileged to utter sedition, to blaspheme, to outrage good taste, to corrupt the youthful mind, and generally to scandalize one`s uncles. - George Bernard Shaw', 0.09048955934140109), ('It is always thus, impelled by a state of mind which is destined not to last, that we make our irrevocable decisions. - Marcel Proust', 0.2262238983535027), ('A mind that is stretched by a new experience can never go back to its old dimensions. - Oliver Wendell Holmes', 0.2262238983535027), ('The mind of the thoroughly well-informed man is a dreadful thing. It is like a bric-à-brac shop, all monsters and dust, with everything priced above its proper value. - Oscar Wilde', 0.2262238983535027), ('No passion so effectually robs the mind of all its powers of acting and reasoning as fear. - Edmund Burke', 0.2262238983535027), ('Too many poets delude themselves by thinking that the mind is dangerous and must be left out. Well, the mind is dangerous and must be left in. - Robert Frost', 0.4524477967070054), ('It is your mind that creates this world. - Siddhartha Buddha', 0.4524477967070054), ('As every divided kingdom falls, so every mind divided between many studies confounds and saps itself. - Leonardo da Vinci', 0.2262238983535027), ('My mind is my own church. - Thomas Paine', 0.2262238983535027), ('What we are today comes from our thoughts of yesterday and our present thoughts build our life tomorrow. Our life is the creation of our mind. - Siddhartha Buddha', 0.09048955934140109), ('Beauty is ever to the lonely mind a shadow fleeting; she is never plain. She is a visitor who leaves behind the gift of grief, the souvenir of pain. - Christopher Morley', 0.15081593223566847), ('What a man`s mind can create, man`s character can control. - Thomas Edison', 0.2262238983535027), ('Games lubricate the body and the mind. - Benjamin Franklin', 0.2262238983535027), ('Everything that used to be a sin is now a disease. - Bill Maher', 3.3933584753025405)])
